sdk/go/pulumi/stack_reference.go

```diff
@@ -53,11 +53,16 @@ func (s *StackReference) GetOutput(name StringInput) AnyOutput {
 
 // GetStringOutput returns a stack output keyed by the given name as an StringOutput
 func (s *StackReference) GetStringOutput(name StringInput) StringOutput {
-	return s.GetOutput(name).ApplyT(func(out interface{}) (string, error) {
-		if s, ok := out.(string); ok {
-			return s, nil
+	return All(name, s.GetOutput(name)).ApplyT(func(args []interface{}) (string, error) {
+		name, out := args[0].(string), args[1]
+		if out == nil {
+			return "", fmt.Errorf("stack reference output %q does not exist on stack %q", name, s.name)
```

Comment:
It's a little weird that `GetOutput` doesn't error on missing keys (which seems intentional from comments), but that these helpers *do* error on missing keys.  It seems we made them do so on purpose in https://github.com/pulumi/pulumi/pull/9855.  So I've left that behaviour the same, and just improved the error messages for now.